kartothek/io/testing/index.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+import pytest
+
+from kartothek.core.factory import DatasetFactory
+from kartothek.core.index import ExplicitSecondaryIndex
+from kartothek.io.eager import (
+    read_dataset_as_metapartitions,
+    store_dataframes_as_dataset,
+)
+
+
+def index_dct_regulator(indices_dict):
```

Comment:
I would expect that the index object equal implementation would handle this already. Isn't this the case? What's missing? Maybe it makes sense to extend the index class with a better `equals` method

Reply:
We are comparing the `index_dct` attribute there. Given that the `ExplicitSecondaryIndex` object may look something like this:
```
ipdb> dataset_factory.indices                                                                     
ExplicitSecondaryIndex(index_storage_key=dataset_uuid/indices/p/2019-05-07T13%3A44%3A31.731906.by-dataset-index.parquet, column=p, dtype=int64, creation_time=2019-05-07 13:44:37.746877, index_dct=[1, 2, 3])
```
Attributes such as `creation_time` and `index_storage_key` are impossible to infer. Thus, I only compare `index_dct`.
We could either discard these time-based attributes in the index `__eq__` method, or (better) change the definition of the `index_dct` attribute to support unordered comparisons

Reply:
The most simple solution may be to just implement a getter method for `IndexBase` which returns the `index_dct` value in the same way as `index_dct_regulator`

Reply:
I'm hesitant to introduce a getter which sorts the attributes by default. These dictionaries and the lists as values may become quite large and sorting the entire dict is quite expensive

Reply:
Please change this to an `assert_index_dct_equal` s.t. we may benefit from better assertion messages. Sorting all dict values should be sufficient and allow python/pytest to use dict assertion messages which are, I believe, better readable than list assertion messages.

A simply way to do this could be done using https://toolz.readthedocs.io/en/latest/api.html#toolz.dicttoolz.valmap

```python
from toolz.dicttoolz import valmap
def assert_index_dct_equal(dict1, dict2):
    dict1 = valmap(dict1, sorted)
    dict2 = valmap(dict2, sorted)
    assert dict1 == dict2
```